amr-wind/turbulence/LES/AMD.cpp

```diff
@@ -0,0 +1,400 @@
+#include <cmath>
+
+#include "amr-wind/turbulence/LES/AMD.H"
+#include "amr-wind/turbulence/TurbModelDefs.H"
+
+#include "AMReX_REAL.H"
+#include "AMReX_MultiFab.H"
+#include "AMReX_ParmParse.H"
+
+namespace amr_wind {
+namespace turbulence {
+
+template <typename Transport>
+AMD<Transport>::AMD(CFDSim& sim)
+    : TurbModelBase<Transport>(sim)
+    , m_vel(sim.repo().get_field("velocity"))
+    , m_temperature(sim.repo().get_field("temperature"))
+    , m_rho(sim.repo().get_field("density"))
+{
+    auto& phy_mgr = this->m_sim.physics_manager();
+    if (!phy_mgr.contains("ABL"))
+        amrex::Abort("AMD model only works with ABL physics");
```

Comment:
Is this true? Is it easy to lift this?

Reply:
I have changed the implementation to be conditional on ABL physics instead of explicitly requiring it. However I'm not sure if having an `alphaeff` function will affect non ABL runs. Should I change that to be 0 explicitly without ABL physics?

Reply:
hmmmm I wouldn't want to break anything... so I think that might be safer.